### PR TITLE
Display total and money request in offline

### PR DESCRIPTION
**Pull request by dukenv0307**
### Details

Display total and money request in offline
### Fixed Issues

$ https://github.com/Expensify/App/issues/26424
PROPOSAL: https://github.com/Expensify/App/issues/26424#issuecomment-1799262109

### Tests

1. Create some money requests in a chat
2. Open the IOU report and send some messages
3. Open another report and send a message
4. Logout and login again with the same account
5. Without opening any other report, go offline
6. Open IOU report that we created above
7. Verify that the total is displayed and at least one money request appears
- [x] Verify that no errors appear in the JS console

### Offline tests

Precondition: Steps 1-3 in online
Steps 4-6
### QA Steps

1. Create some money requests in a chat
2. Open the IOU report and send some messages
3. Open another report and send a message
4. Logout and login again with the same account
5. Without opening any other report, go offline
6. Open IOU report that we created above
7. Verify that the total is displayed and at least one money request appears
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/129500732/d4d1382e-1f74-4845-a569-9fdb76b6bcef

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/129500732/8dc50c6e-f486-413c-a8b1-2ade1cec54a9

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/129500732/f4869237-6974-4c69-8130-8bc7e8e6bc2c

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/129500732/694bd841-aa7b-46b2-a924-bfa6990bcaf7

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/129500732/30377502-4ef3-471c-a804-a8bdfa550e92

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/129500732/82271331-262b-4906-b10e-1ab24d72d5dc

</details>

### CODE REVIEW

**Comment by mountiny** [comment]
@abdulrahuman5196 Hello! What is your ETA for the checklist? thank you!

**Comment by abdulrahuman5196** [comment]
Hi @mountiny , Working on it. Would be completing a day.

**Comment by abdulrahuman5196** [comment]
Reviewing now

File: src/pages/home/report/ReportActionsView.js [hunk 1]
```diff
@@ -245,6 +248,50 @@ function ReportActionsView(props) {
         }
     }, [hasCachedActions]);
 
+    const reportActionsToDisplay = useMemo(() => {
```
Comment (abdulrahuman5196) [inline src/pages/home/report/ReportActionsView.js:251]: @dukenv0307 Can we add clear comments here?
Like `What are we doing here?` and `Why are we doing this?` and link the decision GH comments as required.
Since this is not a straightforward approach and would be helpful for people seeing this at future

**Comment by abdulrahuman5196** [comment]
Did some initial testing. i was able to test the displaying of total. But unable to complete the skeleton loading part. Will update here once done.

**Comment by abdulrahuman5196** [comment]
> @abdulrahuman5196
> 
> > Did some initial testing. i was able to test the displaying of total. But unable to complete the skeleton loading part. Will update here once done.
> 
> Seems like BE always returns the money request action in `OpenApp` API.

@dukenv0307 Could you provide more info on this?

**Comment by abdulrahuman5196** [comment]
@dukenv0307 
Got it. Yeah I think so the same. Will try to see if I can create 100s of requests and see. 


Kindly add videos of iOS and Android native on the author's checklist?
And the skeleton case is not present in your videos as well, do record with that case as well.

File: src/pages/home/report/ReportActionsView.js [hunk 2]
```diff
@@ -245,6 +248,55 @@ function ReportActionsView(props) {
...
+        }
+
+        if (
+            props.report.total &&
+            _.filter(
+                props.reportActions,
+                (action) => action.actionName === CONST.REPORT.ACTIONS.TYPE.IOU && action.originalMessage && action.originalMessage.type === CONST.IOU.REPORT_ACTION_TYPE.CREATE,
+            ).length === 0
+        ) {
+            const optimisticIOUAction = ReportUtils.buildOptimisticIOUReportAction(
```
Comment (abdulrahuman5196) [inline src/pages/home/report/ReportActionsView.js:278]: I tried to test this case locally by removing the if checks, the skeleton comes. But it is not animating like other skeletons do.



https://github.com/Expensify/App/assets/46707890/73434c0b-0b31-4dff-b7ff-b21538e652af

File: src/pages/home/report/ReportActionsView.js [hunk 3]
```diff
@@ -245,6 +248,55 @@ function ReportActionsView(props) {
         }
     }, [hasCachedActions]);
 
+    // When we offline before opening a money request report,
```
Comment (abdulrahuman5196) [inline src/pages/home/report/ReportActionsView.js:251]:
```suggestion
    // When we are offline before opening a money request report,
```

**Review by abdulrahuman5196 — CHANGES_REQUESTED** [review]
@dukenv0307 , could you kindly check on these comments?

File: src/pages/home/report/ReportActionsView.js [hunk 4]
```diff
@@ -245,6 +248,55 @@ function ReportActionsView(props) {
...
+                DateUtils.subtractMillisecondsFromDateTime(_.last(props.reportActions).created, 1),
+            );
+            actions.push(optimisticCreatedAction);
+        }
+
+        if (
+            props.report.total &&
+            _.filter(
+                props.reportActions,
+                (action) => action.actionName === CONST.REPORT.ACTIONS.TYPE.IOU && action.originalMessage && action.originalMessage.type === CONST.IOU.REPORT_ACTION_TYPE.CREATE,
```
[the remark this answers is not in the document]
Comment (abdulrahuman5196) [reply src/pages/home/report/ReportActionsView.js:275]: >Is it true that we will add 1 more request action if the current number of actions is not equal to the total number of requests?

Yes. Exactly. We are checking if there is report action with `CONST.IOU.REPORT_ACTION_TYPE.CREATE` type right? Or kindly correct me if I misunderstood the logic of the code.

**Comment by mountiny** [comment]
@abdulrahuman5196 Can you confirm with the OpenApp response that the reportActions collection is returned?

I think that what you are seeing might be the case because of all the IOU report actions being parent report actions for the transaction threads and we do return all of those on OpenApp. So if that is what you are seeing it might be the case

We do not return them for IOU reports though so you could make IOU p2p requests and test there

File: src/pages/home/report/ReportActionsView.js (hunk 2 above)
Comment (abdulrahuman5196) [reply src/pages/home/report/ReportActionsView.js:281]: @dukenv0307 

>The reason is MoneyRequestSkeletonView has backgroundColor and foregroundColor are the same, no animation appears.

Is it even previously we didn't have animation where the money request skeleton was display? 🤔 
I think I remember seeing animation, not sure though

**Comment by abdulrahuman5196** [comment]
Reviewing now

**Comment by abdulrahuman5196** [comment]
@dukenv0307 I am seeing issue, where some of the transactions and comments are missing inbetween, but not showing any loader in screen while offline. After going online only the issue is getting fixed. 


https://github.com/Expensify/App/assets/46707890/cbe865d3-888b-4350-80f5-27dd71edc31e

**Comment by abdulrahuman5196** [comment]
@dukenv0307 these are all already available requests. You could see dates of the request from the video I suppose

**Comment by abdulrahuman5196** [comment]
> @abdulrahuman5196 So the steps is this, right?
> 
>     1. Created all requests
> 
>     2. Sign out and sign again
> 
>     3. Offline
> 
>     4. No skeleton is displayed

@dukenv0307 
for step 1, these are pre-existing reports in account, created at different times. Other steps are same.

The weird fact is some comments and transactions are missing in-between as shown in the video

**Comment by abdulrahuman5196** [comment]
@dukenv0307 Can you check with adding multiple requests like around 50 or 30 requests? I am not seeing consistent working on the below cases. Attaching the video.
1) Didn't show the total amount for one of my chat.
2) Didn't show description for money requests in workspace chat


https://github.com/Expensify/App/assets/46707890/4bc09752-e796-4696-86e1-e7ba67ee7ae9

**Comment by abdulrahuman5196** [comment]
> @abdulrahuman5196
> 
> > The weird fact is some comments and transactions are missing in-between as shown in the video
> 
> In `OpenApp` API, the comments actions will not be returned and for P2P requests, we only return some of the transactions
> Screen.Recording.2024-03-04.at.17.53.35.mov
> 
> > Didn't show the total amount for one of my chats.
> 
> With an account that has many reports, some P2P IOU reports can have no action in `OpenApp` API which is the reason of this case
> 
> > Didn't show description for money requests in workspace chat
> 
> For this case, what is the description for money requests?

Thanks for the update @dukenv0307 .



@mountiny I also checked and it seems in OpenApp API we send less reportActions for P2P reports, but the problem is the reportActions are not in order. Meaning reportActions/transaction are missing in between transactions. I tested with user having 50+ requests - https://github.com/Expensify/App/pull/36657#issuecomment-1966046900

Can we kindly check it in backend? If its bug in backend, we should fix that one bug in backend.
Since if we make the skeleton change and show all the transactions, user could find some transactions missing inbetween.


And to note merchant information is also not return in OpenAPI, which is an existing issue even now.

**Comment by mountiny** [comment]
Sorry I have been ooo and getting a bit lost with the state of this PR, could you summarize what you think are the current blockers here and how I can help?

**Comment by mountiny** [comment]
What is the specific case the transaction is missing?

**Comment by mountiny** [comment]
We do not send all the P2P transaction threads or report actions presumably when there were no comments on it. 

Could we just simplify this as much as possible and show that one skeleton for expense reports if the report actions are missing? 

Regarding the test steps:

> Open IOU report that we created above
Verify that the total is displayed and at least one money request appears 

We do not send the IOU (P2P) reports when there are no comments (notification preference is hidden), this is intentional and I dont think we would change that behaviour

**Comment by abdulrahuman5196** [comment]
> Could we just simplify this as much as possible and show that one skeleton for expense reports if the report actions are missing?

Got it.

**Comment by abdulrahuman5196** [comment]
I think I have already looked into the code. I will work on platform videos in my morning.

**Comment by abdulrahuman5196** [comment]
Sorry for the delay working on review videos now.

File: src/pages/home/report/ReportActionsView.tsx [hunk 5]
```diff
@@ -221,6 +224,61 @@ function ReportActionsView({
...
+                DateUtils.subtractMillisecondsFromDateTime(actions[actions.length - 1].created, 1),
+            ) as OnyxTypes.ReportAction;
+            moneyRequestActions.push(optimisticIOUAction);
+            actions.splice(actions.length - 1, 0, optimisticIOUAction);
+        }
+
+        // Update pending action of created action if we have some requests that are pending
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const createdAction = actions.pop()!;
+        if (moneyRequestActions.filter((action) => Boolean(action.pendingAction)).length > 0) {
```
Comment (abdulrahuman5196) [inline src/pages/home/report/ReportActionsView.tsx:275]: @dukenv0307 
I am getting an issue where even in online state, if i can create a request the total has reduced opacity.

This should be case only for offline. Since even the pending requests shows in lesser opacity, the total should behave the same way.


https://github.com/Expensify/App/assets/46707890/e4e9ea25-7ea7-4213-a459-39b0b0beae73

**Review by abdulrahuman5196 — CHANGES_REQUESTED** [review]

File: src/pages/home/report/ReportActionsView.tsx (hunk 5 above)
Comment (abdulrahuman5196) [reply src/pages/home/report/ReportActionsView.tsx:453]: Can you try with slow network, it occurs till the money request API succeeds.

**Comment by abdulrahuman5196** [comment]
checking again since we made a merge from live

**Comment by abdulrahuman5196** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>





https://github.com/Expensify/App/assets/46707890/7263a806-82e1-4104-8eed-36165edc2635



</details>

<details>
<summary>Android: mWeb Chrome</summary>




https://github.com/Expensify/App/assets/46707890/4fd5737f-cbd6-4386-a8d1-2576c1c49ec4




</details>

<details>
<summary>iOS: Native</summary>






https://github.com/Expensify/App/assets/46707890/0c40acf9-0cf2-4a57-b6b6-adab1856dd0c




</details>

<details>
<summary>iOS: mWeb Safari</summary>





https://github.com/Expensify/App/assets/46707890/ff78902f-f969-46d1-bce3-24a6e1ec9712



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>





https://github.com/Expensify/App/assets/46707890/dac92cda-e7cb-4bd1-a9db-6924d1132180


https://github.com/Expensify/App/assets/46707890/2e508381-3510-4308-8d3e-2bbc908d0c3b


</details>

<details>
<summary>MacOS: Desktop</summary>





https://github.com/Expensify/App/assets/46707890/ce0b285c-0cee-4a5d-a6cb-cb0530b3e788



</details>

**Review by abdulrahuman5196 — APPROVED** [review]
Changes looks good and works well. Reviewers checklist is also complete.

All yours. @mountiny 

🎀 👀 🎀
C+ Reviewed

File: src/pages/home/report/ReportActionsView.tsx [hunk 6]
```diff
@@ -409,6 +412,61 @@ function ReportActionsView({
         };
     }, [isTheFirstReportActionIsLinked]);
 
+    // When we are offline before opening a money request report,
+    // the total of the report and sometimes the money request aren't displayed because these actions aren't returned until `OpenReport` API is complete.
+    // We generate a fake created action here if it doesn't exist to display the total whenever possible because the total just depends on report data
+    // and we also generate a money request action if If the number of money requests in reportActions is less than the total number of money requests
```
Comment (mountiny) [inline src/pages/home/report/ReportActionsView.tsx:418]:
```suggestion
    // and we also generate a money request action if the number of money requests in reportActions is less than the total number of money requests
```

File: src/pages/home/report/ReportActionsView.tsx [hunk 7]
```diff
@@ -409,6 +412,61 @@ function ReportActionsView({
...
+    // to display at least one money request action to match the total data.
+    // For more details: https://github.com/Expensify/App/issues/26424#issuecomment-1869154198
```
Comment (mountiny) [inline src/pages/home/report/ReportActionsView.tsx:420]: This can be found from blame
```suggestion
```

**Review by mountiny — COMMENTED** [review]
@abdulrahuman5196 @dukenv0307 in the videos here 
image
 shouldnt there be a skeleton? It is on web/desktop

**Review by mountiny — APPROVED** [review]
Alright, thanks for the update, lets get this merged now
